librz/core/p/core_java.c

```diff
@@ -10,1659 +9,321 @@
-			rz_cons_printf(" [ NONE ]\n");
-		} else {
-			rz_cons_printf(" Exception table for %s (%d entries) @ 0x%" PFMT64x ":\n", method->name,
-				rz_list_length(exc_table), start);
-		}
-		rz_list_foreach (exc_table, exc_iter, exc_entry) {
-			char *class_info = rz_bin_java_resolve_without_space(obj, exc_entry->catch_type);
-			rz_cons_printf("  Catch Type: %d, %s @ 0x%" PFMT64x "\n", exc_entry->catch_type,
-				class_info, exc_entry->file_offset + 6);
-			rz_cons_printf("  Start PC: (0x%x) 0x%" PFMT64x " @ 0x%" PFMT64x "\n",
-				exc_entry->start_pc, exc_entry->start_pc + start, exc_entry->file_offset);
-			rz_cons_printf("  End PC: (0x%x) 0x%" PFMT64x " 0x%" PFMT64x "\n",
-				exc_entry->end_pc, exc_entry->end_pc + start, exc_entry->file_offset + 2);
-			rz_cons_printf("  Handler PC: (0x%x) 0x%" PFMT64x " 0x%" PFMT64x "\n",
-				exc_entry->handler_pc, exc_entry->handler_pc + start, exc_entry->file_offset + 4);
-			free(class_info);
-		}
-	}
-	return RZ_CMD_STATUS_OK;
-}
-
-RZ_IPI RzCmdStatus rz_cmd_java_handler(RzCore *core, int argc, const char **argv) {
-	if (argc < 2) {
-		return RZ_CMD_STATUS_WRONG_ARGS;
-	}
-	for (int i = 0; i < END_CMDS; i++) {
-		if (!strncmp(argv[1], JAVA_CMDS[i].name, JAVA_CMDS[i].name_len)) {
-			return JAVA_CMDS[i].handler(core, argc - 2, argc > 2 ? &argv[2] : NULL);
-		}
-	}
-	return RZ_CMD_STATUS_WRONG_ARGS;
-}
-
-static const char *cmd_java_subcmd_choices[] = { "help", "set_flags", "prototypes", "resolve_cp", "calc_flags", "flags_str_at", "flags_str", "m_info", "f_info", "find_cp_const", "find_cp_value", "replace_cp_value", "replace_classname_value", "reload_bin", "summary", "exc", "calc_sz", "is_valid", NULL };
-static const RzCmdDescArg cmd_java_args[3] = {
-	{
-		.name = "subcmd",
-		.type = RZ_CMD_ARG_TYPE_CHOICES,
-		.default_value = "help",
-		.choices = cmd_java_subcmd_choices,
-	},
+static const RzCmdDescArg name_args(javar)[] = {
 	{
-		.name = "arg",
-		.type = RZ_CMD_ARG_TYPE_STRING,
-		.flags = RZ_CMD_ARG_FLAG_ARRAY,
-		.optional = true,
+		.name = "index",
+		.type = RZ_CMD_ARG_TYPE_NUM,
 	},
 	{ 0 },
 };
 
-static const RzCmdDescHelp cmd_java_help = {
-	.summary = "Suite of java commands",
-	.description = "Type `java help` for more commands.",
-	.args = cmd_java_args,
+static const RzCmdDescHelp name_help(javar) = {
+	.summary = "resolves the class constant pool value at a given index",
+	.args = name_args(javar),
 };
 
 static bool rz_cmd_java_init_handler(RzCore *core) {
-	RzCmd *cmd = core->rcmd;
-	RzCmdDesc *root_cd = rz_cmd_get_root(cmd);
+	RzCmd *rcmd = core->rcmd;
+	RzCmdDesc *root_cd = rz_cmd_get_root(rcmd);
 	if (!root_cd) {
 		return false;
 	}
 
-	RzCmdDesc *cmd_java_cd = rz_cmd_desc_argv_new(cmd, root_cd, "java", rz_cmd_java_handler, &cmd_java_help);
-	rz_warn_if_fail(cmd_java_cd);
-	return cmd_java_cd != NULL;
+	RzCmdDesc *java = rz_cmd_desc_group_new(rcmd, root_cd, "java", name_handler(java), &name_help(java), &java_usage);
```

Comment:
we don't usually have a command that shows nothing but its help, so I'd remove the handler of the `java` command. One has to do `java?`, for consistency, to get an help.

Reply:
actually i feel like is better to do this way, so when you type java it will prompt you the help message.

Reply:
If we want this behaviour, let's do it for *all* commands in a separate PR. For example, see `zf`. `zf` does not exist, but it is the group name for `zfd` and `zfs`. If we want `zf` to get the help of the `zf` group, then we should have this in RzCmd to ensure consistency, not ad-hoc IMO. Otherwise we won't be consistent.

Reply:
resolved like for zf. i wasn't aware of that way of having a group.